src/tds/stream/query.rs

```diff
@@ -66,17 +66,16 @@ impl<'a> QueryStream<'a> {
 
                     return Ok(());
                 }
-                Some(ReceivedToken::DoneInProc(_)) | Some(ReceivedToken::DoneProc(_)) => {
-                    return Ok(());
-                }
-                Some(ReceivedToken::Done(done)) if done.has_more() => return Ok(()),
-                _ => {
-                    if self.columns().is_none() {
+                Some(ReceivedToken::DoneInProc(done))
+                | Some(ReceivedToken::DoneProc(done))
+                | Some(ReceivedToken::Done(done)) => {
+                    if !done.has_more() {
```

Comment:
So if I understand this correctly, this is the part about swallowed errors?

Reply:
Yeah. We must go through the whole stream always. We might get any of the done tokens, but they have a bitflag set if there's more data to be come.